### PR TITLE
Updated the github workflow to generate the JSON schemas and Definitions for OSM components

### LINKED ISSUE #93: [CI] Patterns: Generate JSON Schema and Definitions for OSM components
**Enhancement Description**

To generate the workflow of getting the manifest from the OSM using the [kubeopenapi-jsonschema](https://github.com/layer5io/kubeopenapi-jsonschema) to read and generate json schemas from those manifests and then also generating workload definitions from those schemas

A sample workflow is [here](https://github.com/DelusionalOptimist/meshery-istio/blob/gen-oam/.github/workflows/update-oam-defs.yml)

---

### PR DESCRIPTION
**Pull request by sayantan1413**
**Description**

Updated the github workflow to generate JSON Schema and Definitions for OSM components

This PR fixes #93 

**Notes for Reviewers**

Continued the #97 

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by sayantan1413** [comment]
Generated templates: https://github.com/sayantan1413/meshery-osm/tree/master/templates/oam/workloads/workloads-v0.9.1
Schemas: https://github.com/sayantan1413/meshery-osm/blob/master/templates/oam/workloads/workloads-v0.9.1/egress.osm.meshery.layer5io.schema.json#L103
Workload defs: https://github.com/sayantan1413/meshery-osm/blob/master/templates/oam/workloads/workloads-v0.9.1/egress.osm_definition.json#L11

**Comment by leecalcote** [comment]
Gents, in lieu of our recent conversations, are we proceeding here (even if only temporarily to produce a round of components)?

**Comment by DelusionalOptimist** [comment]
@sayantan1413 we're close, just some small changes according to [this](https://github.com/meshery/meshery-istio/blob/master/.github/workflows/update-oam-defs.yml) (finally :P)
Let's have this in and get these components generated for OSM. :muscle:

**Comment by sayantan1413** [comment]
Hey @DelusionalOptimist can you verify this https://github.com/sayantan1413/meshery-osm/tree/master/templates/oam/workloads/workloads-v0.9.1 ?

File: .github/workflows/update-oam-defs.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          cd util
+          npm i; npm i nexe -g
+          make linux
+      - name: Generate Workload definitions
+        run: |
+          dirPath=../templates/oam/workloads/workloads-$NEW_OSM_VERSION
+          cd util
+          mkdir -p $dirPath
+          meshName=osm
+          smpMeshName=OSM
```
Comment (DelusionalOptimist) [inline .github/workflows/update-oam-defs.yml:70]:
```suggestion
          smpMeshName=OPEN_SERVICE_MESH
```
SMP likes it : )

File: .github/workflows/update-oam-defs.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            | ."spec"."metadata"."k8sKind" = "'$nameUpper'"
+            | ."spec"."definitionRef"."name"="'$definitionRef'"' > $dirPath/$nameLower.osm_definition.json
+          done
+      - name: Cleanup
+        run: |
+          rm -rf util
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          file_pattern: ./templates ./.github
```
Comment (DelusionalOptimist) [inline .github/workflows/update-oam-defs.yml:98]:
```suggestion
          file_pattern: templates/
```
We don't need to add anything to `.github` now.

**Review by DelusionalOptimist — APPROVED** [review]
Looking good to go :+1:.
Great work @sayantan1413 :rocket:

**Comment by sayantan1413** [comment]
Thank you @DelusionalOptimist for your constant support 🥳

**Pull request merged**